qiita_db/util.py

```diff
@@ -1549,3 +1551,86 @@ def generate_study_list(study_ids, build_samples, public_only=False):
             infolist.append(info)
 
     return infolist
+
+
+def generate_biom_and_metadata_release(study_status):
+    """Generate a list of biom/meatadata filepaths and a tgz of those files
+
+    Parameters
+    ----------
+    study_status : str
+        The study status to search for. Note that this should always be set
+        to 'public' but having this expose helps with testing
+
+    Returns
+    -------
+    str, str
+        tgz_name: the filepath of the new generated tgz
+        txt_name: the filepath of the new generated txt
+    """
+    studies = qdb.study.Study.get_by_status(study_status)
+    qiita_config = ConfigurationManager()
+    working_dir = qiita_config.working_dir
+    portal = qiita_config.portal
+    bdir = qdb.util.get_db_files_base_dir()
+    bdir_len = len(bdir) + 1
+
+    data = []
+    for s in studies:
+        for a in s.artifacts():
+            if a.artifact_type == 'BIOM':
+                if a.processing_parameters is None:
+                    continue
+
+                # [0] latest is first, [1] only getting the filepath
+                sample_fp = a.study.sample_template.get_filepaths()[0][1]
+                if sample_fp.startswith(bdir):
+                    sample_fp = sample_fp[bdir_len:]
+                cmd_name = a.processing_parameters.command.name
+
+                # this loop is necessary as in theory an artifact can be
+                # generated from multiple prep info files
+                human_cmd = []
+                for p in a.parents:
+                    pp = p.processing_parameters
+                    pp_cmd_name = pp.command.name
+                    if pp_cmd_name == 'Trimming':
```

Comment:
I _guess_ this _works_, but this is hardcoding something to a plugin value... The whole point of plugins was to avoid this kind of things on the Qiita code...